HIP/hip-0000-add-software-version-to-event.md

```diff
@@ -0,0 +1,111 @@
+---
+hip: <HIP number (this is determined by the HIP editor)>
+title: Add Software Version to Events
+author: Edward Wertz <@edward-swirldslabs>
+working-group: a list of the technical and business stakeholders' name(s) and/or username(s), or name(s) and email(s).
+type: Standards Track
+category: Core
+needs-council-approval: <Yes | No>
+status: Draft
+created: 4/19/2023
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/720
+updated: <comma separated list of dates>
+requires: <HIP number(s)>
+replaces: <HIP number(s)>
+superseded-by: <HIP number(s)>
+---
+
+## Abstract
+
+Add the software version to events that are serialized and to the event interface. This permits the application to
+determine which version of software was used to create each event, and optionally to ignore events created by previous
+versions.
+
+## Motivation
+
+This change should come before the implementation of the preconsensus event stream. The preconsensus event stream
+will save events to disk prior to their reaching consensus. On restart of the application all events in the preconsensus
+event stream will be replayed to reach consensus. If the restart coincides with a software upgrade, the new software
+version will receive events created from the previous software version. This change would enable the application to
+decide whether to act on the transactions generated by the older software version.
+
+## Rationale
+
+Currently, events which do not reach consensus prior to a software upgrade are lost and the transactions are never
+handled. With the introduction of the preconsensus event stream, replayed events may reach consensus and the
+transactions would become available for handling. Adding the version number to the event will allow the application
+to decide how to handle the older transactions. Software upgrades can take several minutes and transactions submitted
+prior to the software upgrade may have timed out. [1](https://github.com/hashgraph/hedera-services/issues/6026)
+
+If this feature is not implemented, the application will not know when it is receiving events that have been generated
+by a different software version and would attempt to handle them in the same way.
+
+## User stories
+
+* John submits a transaction just as the mainnet undergoes a software upgrade. When John comes back after the
+  software upgrade to check on the status of his transaction, he can see whether the transaction came to
+  consensus and was handled. If the transaction failed to be handled, John can be given more specific reasons as to
+  why.
```

Comment:
I would just write "This is an internal platform feature, and will not be something that users will directly interact with."

Reply:
I disagree.  I think the ability to have transactions come to consensus and be handled after an upgrade is user impacting.

Reply:
I've added your text before the user story.